File: app/src/main/java/com/molihua/hlbmerge/fragment/AbstractMainFragment.java
package com.molihua.hlbmerge.fragment;

import android.content.Context;

import androidx.annotation.NonNull;

import com.molihua.hlbmerge.activity.AbstractMainActivity;

/**
 * @ClassName: AbstractMainFragment
 * @Author: molihuan
 * @Date: 2022/12/20/17:14
 * @Description:
 */
public abstract class AbstractMainFragment extends BaseFragment {
    protected AbstractMainActivity abstractMainActivity;

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        if (mActivity instanceof AbstractMainActivity) {
            abstractMainActivity = (AbstractMainActivity) mActivity;
        }
    }
}

File: app/src/main/java/com/molihua/hlbmerge/fragment/AbstractMainFragment.kt
package com.molihua.hlbmerge.fragment;

import android.content.Context
import androidx.viewbinding.ViewBinding
import com.molihua.hlbmerge.activity.AbstractMainActivity

/**
 * @ClassName: AbstractMainFragment
 * @Author: molihuan
 * @Date: 2022/12/20/17:14
 * @Description:
 */
abstract class AbstractMainFragment : BaseFragment() {
    @JvmField
    protected var abstractMainActivity: AbstractMainActivity<ViewBinding>? = null

    override fun onAttach(context: Context) {
        super.onAttach(context)
        val activity = getActivity();
        if (activity is AbstractMainActivity<*>) {
            abstractMainActivity = activity as AbstractMainActivity<ViewBinding>
        }
    }
}
